x/shield/keeper/purchase.go

```diff
@@ -188,6 +187,8 @@ func (k Keeper) RemoveExpiredPurchasesAndDistributeFees(ctx sdk.Context) {
 
 				// If purchaseDeletionTime < currentBlockTime, remove the purchase.
 				if entry.DeletionTime.Before(ctx.BlockTime()) {
+					k.DequeuePurchase(ctx, purchaseList, entry.ProtectionEndTime)
```

Comment:
Consider this case:
PurchaseList1
- entry1: completed on Day 1; deleted on Day 1 (no claim)
- entry2: completed on Day 1; deleted on Day 5 (claim)
Then on Day 1 would entry2 also be removed?

Reply:
Yeah this could be problematic just like unbonding entries.. 🤔

Reply:
Oh I guess unbonding entries are fine now because we don’t deal with them in a problematic manner. What do you think about dequeueing only after protection end time + voting period has passed? This was my initial approach, but I switched due to import export errors.

Reply:
Actually, I will try to come up with a better solution

Reply:
I will see if there is a better solution

Reply:
Actually `DequeuePurchase`, which I renamed to `DequeuePoolPurchaser`, only removes one element and exits. And our queue stores duplicates. So I think it's fine.

Reply:
Actually `DequeuePurchase`, which I renamed to `DequeuePoolPurchaser`, only removes one element and terminates. And our queue stores duplicates. So I think it's fine.

Reply:
Actually `DequeuePurchase` only removes one element and terminates, while our queue stores duplicate pool-purchaser pairs. So I think it's fine.

Reply:
Yeah you're right.